packages/phone-number-privacy/combiner/test/integration/pnp.test.ts

```diff
@@ -444,4 +454,192 @@ describe('pnpService', () => {
       })
     })
   })
+
+  describe(`${CombinerEndpoint.PNP_QUOTA}`, () => {
+    const useQuery = async (queryCount: number, signer: Server | HttpsServer) => {
+      for (let i = 0; i < queryCount; i++) {
+        const phoneNumber = '+1' + Math.floor(Math.random() * 10 ** 10)
+        const blindedNumber = getBlindedPhoneNumber(phoneNumber, BLINDING_FACTOR)
+        const req = {
+          account: ACCOUNT_ADDRESS1,
+          blindedQueryPhoneNumber: blindedNumber,
+        }
+        const authorization = getPnpRequestAuthorization(req, ACCOUNT_ADDRESS1, PRIVATE_KEY1)
+        await request(signer)
+          .post(SignerEndpoint.PNP_SIGN)
+          .set('Authorization', authorization)
+          .send(req)
+      }
+    }
+
+    const getCombinerQuotaResponse = async (req: PnpQuotaRequest, authorization: string) => {
+      const res = await request(app)
+        .get(CombinerEndpoint.PNP_QUOTA)
+        .set('Authorization', authorization)
+        .send(req)
+      return res
+    }
+
+    const totalQuota = 10
+    beforeAll(async () => {
+      let weiTocusd = new BigNumber(1e17)
+      mockOdisPaymentsTotalPaidCUSD.mockReturnValue(weiTocusd.multipliedBy(totalQuota))
+    })
+
+    const queryCountParams = [
+      [0, 0, 0, 0],
+      [1, 0, 0, 0], // does not reach threshold
+      [1, 1, 0, 1], // threshold reached
+      [0, 1, 1, 1], // order of signers shouldn't matter
+      [1, 4, 9, 4],
+    ]
+    queryCountParams.forEach(
+      ([signer1queries, signer2queries, signer3queries, expectedQueryCount]) => {
+        it(`should get ${expectedQueryCount} performedQueryCount given signer responses of [${signer1queries}, ${signer2queries}, ${signer3queries}]`, async () => {
+          await useQuery(signer1queries, signer1)
+          await useQuery(signer2queries, signer2)
+          await useQuery(signer3queries, signer3)
+
+          const req = {
+            account: ACCOUNT_ADDRESS1,
+          }
+          const authorization = getPnpRequestAuthorization(req, ACCOUNT_ADDRESS1, PRIVATE_KEY1)
+          const res = await getCombinerQuotaResponse(req, authorization)
+
+          expect(res.body).toMatchObject<PnpQuotaResponseSuccess>({
+            success: true,
+            version: expectedVersion,
+            performedQueryCount: expectedQueryCount,
+            totalQuota,
+            blockNumber: testBlockNumber,
+          })
+        })
+      }
+    )
+
+    // TODO: INSUFFICIENT SIGNER RESPONSES
```

Comment:
Flagging this TODO. Do you plan on doing this as part of this PR?

Reply:
Nvm I can see that you already did!

Reply:
oops, will delete the comment!